### ISSUE TITLE
Allow baseline snapping to a fraction of the grid

### ISSUE THREAD

**Issue by tbredin** [StudioThick/megatype#28]
With a grid such as 16px it's often not viable for small type sizes to sit on this grid; in design this is usually handled by letting every second line (or 3rd, 4th, 5th) sit on the baseline grid. This would be a great feature to have!

Considerations:
- this messes up the flow of sequential elements, do we need to handle a correction, or is this too complicated?

**Comment by scottromack** [comment]
+1  I think the most common usage is every other line which would mean just setting your baseline grid to one half (8px) and all your margins to .5rem or 1rem...?

**Comment by tbredin** [comment]
Yep, that's how you can achieve it manually right now. It feels pretty gross though if you ask me, 8px (`html { font-size: 50%; }`) is kinda weird. Would prefer to implement a config variable perhaps, something like `$baseline-subdivisions: 1 !default;` which you could, for example, set to `$baseline-subdivisions: 2;` to allow the typeset mixin to round to `0.5rem` instead of `1rem` as it currently does.

**Comment by tbredin** [comment]
Added in `1.1.1`.

```
// baseline subdivisions specify a fraction of the grid that type can be snapped to.
// useful for small text
// can be overridden in typeset mixin with parameter $use-subdivisions: false, $use-subdivisions: 3, etc
$baseline-subdivisions: 1;
```

Note that as the most common use for this is with small text this can interact strangely with pixel rounding. For best results ensure that your rootsize is divisible by the subdivision you specify, eg: subdivisions 3 will work well with a `rootsize` of 12, 15, 18, but not 10, 14, etc.

I also added a warning to the sass output to warn users if they do this.
